qcodes/dataset/sqlite/queries.py

```diff
@@ -225,6 +269,13 @@ def get_parameter_data_for_one_paramtree(
 
     for paramspec, column_data in zip(paramspecs, res_t):
         try:
+            if paramspec.type == "numeric":
+                # there is no reliable way to
+                # tell the difference between a float and and int loaded
+                # from sqlite numeric columns so always fall back to float
+                dtype: Optional[type] = np.float64
+            else:
```

Comment:
Nope this is only requied because there is no way to tell the difference between float and ints storred as the numeric type in sqlite. Complex is storred as its own type. 

For numeric you could have (1,1.5.,2) and so on as the data read back. You will therefor get an array of integers if you only look at the first data point. Which can happen as you read the data back.

Reply:
aah, ok, tjhanks